src/btree/bt_vrfy.c

```diff
@@ -434,6 +434,28 @@ __verify_tree(
         WT_RET(__wt_debug_page(session, NULL, ref, NULL));
 #endif
 
+    /* Make sure the page we got belongs in this kind of tree. */
```

Comment:
I consciously put it after the debug calls, because if anyone ever actually hits it they'll want that dump, and the debug code ought to be able to handle whatever page type it is. At this point it's passed vrfy_dsk so it can't be completely corrupted and the debug code shouldn't choke on it. I don't see anything further above that using anything other than page->type. (Am I missing something?)

Reply:
Yes, the page is not corrupted at this stage. I agree that it will be helpful if someone wants to see that dump.